crates/primitives/src/block.rs

```diff
@@ -204,72 +152,83 @@ impl<'a> arbitrary::Arbitrary<'a> for Block {
 
 /// Sealed block with senders recovered from transactions.
 #[derive(Debug, Clone, PartialEq, Eq, Default, Deref, DerefMut)]
-pub struct BlockWithSenders {
+pub struct BlockWithSenders<B: reth_primitives_traits::Block = Block> {
     /// Block
     #[deref]
     #[deref_mut]
-    pub block: Block,
+    pub block: B,
     /// List of senders that match the transactions in the block
     pub senders: Vec<Address>,
 }
 
-impl BlockWithSenders {
+impl<B: reth_primitives_traits::Block> BlockWithSenders<B> {
```

Comment:
`BlockPrimitives` would have been ideal here, then you have the transaction type accessible without fully qualified syntax and without making yet another adapter type
https://github.com/paradigmxyz/reth/pull/12721 @mattsse